algorithms_task_2/student/src/impl.cpp

```diff
@@ -1 +1,15 @@
 #include "impl.h"
+#include <algorithm>
+
+bool startsWith(const std::string& source, const std::string& prefix)
+{
+    if (prefix.empty()) {
+        return true;
+    }
+
+    if (prefix.size() > source.size()) {
+        return false;
+    }
+
+    return std::equal(source.begin(), source.begin() + prefix.size(), prefix.begin());
```

Comment:
_what if source will be empty?
operator + will result UB for end iterator

looks like it is safe here, but I'd make it safer without any shifts that possibly may result an error_

Reply:
Done!